engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/ExecutionStore.scala

```diff
@@ -44,6 +44,9 @@ final case class ExecutionStore(private val statusStore: Map[JobKey, ExecutionSt
     (key.scope.fullyQualifiedName, key.index) -> key
   } toMap
 
+  lazy val terminalKeys: Set[FqnIndex] = store.filterKeys(_.isTerminal).values.flatten.map { key =>
```

Comment:
+1000 benchmarking any changes here 😄 

is it worth `lazy`ing that `keys map toMapEntry` to avoid calculating it at all if it's not needed?